### DISCUSSION TITLE
[homeassistant.components.switch] Platform esphome does not generate unique IDs.

### DISCUSSION THREAD

**Discussion by cobradodo** [indomus/forum#214] [HUB: Home Assistant]
Nel log di HA ricevo questo errore. Ho provato a:

- eliminare i file relativi a esphome in .storage
- rimuovere l'integrazione e rimetterla
- eliminato il DB
- cercato lo switch in questione nei vari file sotto la .storage ed eliminato le voci riguardanti quello switch

Cercando in giro non trovo soluzioni al mio problema.
A qualcuno è già capitato?

**Comment by snakuzzo** [comment]
1. il DB credo c'entri poco o nulla
2. Verifica di non avere conflitti con le altre entità di tipo switch
3. Non dovresti andare a manella sotto .storage a fare modifiche

Ad ogni modo i rimuoverei l'integrazione e mi assicurerei di rimuovere tutti i device da Configuration > Entities, riavvierei HA e poi aggiungerei nuovamente l'integrazione.

**Reply by cobradodo** [reply]
Eliminando l'integrazione, nell'elenco delle entità non trovo più nulla riguardante quell'oggetto.
Comunque ora ho eliminato l'integrazione e riavviato (ma l'avevo già fatto)

**Reply by cobradodo** [reply]
Infatti... appena lo aggiungo....
![image](https://user-images.githubusercontent.com/57904707/116006238-92572300-a60a-11eb-8c32-3bb89b8716c7.png)

ho ricercato in tutti i file di configurazione e non dichiaro da nessuna parte uno switch del genere...

Hai qualche idea?

**Reply by snakuzzo** [reply]
No...ma di certo non se lo inventa lui...non è che tu stai esponendo dei device che hanno nomi uguali e proprio in fase di discovery ti fa casini ?

**Reply by cobradodo** [reply]
Trovato....
Dalla tua frase "stai esponendo device con nomi uguali"... ho dato uno sguardo al firmware esphome perchè non capivo...e guarda qua...
![image](https://user-images.githubusercontent.com/57904707/116010104-5aa5a680-a61d-11eb-9255-69f124d4a8ef.png)


Maledetto copia e incolla mi sono copiato 2 volte la stessa cosa da un altro mio firmware
